clients/fossid-webapp/src/main/kotlin/model/result/MatchType.kt

```diff
@@ -26,5 +26,12 @@ enum class MatchType {
     IGNORED,
     INTAKE,
     NONE,
-    PARTIAL
+    PARTIAL;
+
+    companion object {
+        /**
+         * Test if the match type is a valid one i.e. the scanner can work with it.
+         */
+        fun MatchType.isValidType() = this == PARTIAL || this == FULL
```

Comment:
Probably my fault, as I believe I've mentioned the term "companion object" before, but this should just be a regular function on the enum itself:

    fun isValidType() = this == FULL || this == PARTIAL

Nit: I've also swapped the order of values to check to make them match the definition order.

Reply:
Also strictly speaking, the introduction of this helper function is a refactoring that could have happened independently before. While I'm fine with keeping it in this commit, the commit message could at least say something like

> While at it, introduce a helper function for valid match types.